test/e2e/framework/helper/certificates.go

```diff
@@ -235,6 +234,83 @@ func (h *Helper) WaitCertificateIssuedValid(ns, name string, timeout time.Durati
 	return h.WaitCertificateIssuedValidTLS(ns, name, timeout, nil)
 }
 
+func (h *Helper) defaultKeyUsagesToAdd(ns string, issuerRef *cmmeta.ObjectReference, keyType cmapi.KeyAlgorithm) (x509.KeyUsage, []x509.ExtKeyUsage, error) {
+	var issuerSpec *cmapi.IssuerSpec
+	switch issuerRef.Kind {
+	case "ClusterIssuer":
+		issuerObj, err := h.CMClient.CertmanagerV1alpha2().ClusterIssuers().Get(issuerRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return 0, nil, fmt.Errorf("failed to find referenced ClusterIssuer %v: %s",
+				issuerRef, err)
+		}
+
+		issuerSpec = &issuerObj.Spec
+	default:
+		issuerObj, err := h.CMClient.CertmanagerV1alpha2().Issuers(ns).Get(issuerRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return 0, nil, fmt.Errorf("failed to find referenced Issuer %v: %s",
+				issuerRef, err)
+		}
+
+		issuerSpec = &issuerObj.Spec
+	}
+
+	var keyUsages x509.KeyUsage
+	var extKeyUsages []x509.ExtKeyUsage
+
+	// Vault and ACME issuers will add server auth and client auth extended key
+	// usages by default so we need to add them to the list of expected usages
+	if issuerSpec.ACME != nil || issuerSpec.Vault != nil {
+		extKeyUsages = append(extKeyUsages, x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth)
+	}
+
+	// Vault issuers will add key agreement key usage
+	if issuerSpec.Vault != nil {
+		keyUsages |= x509.KeyUsageKeyAgreement
+	}
+
+	// Venafi issue adds server auth key usage
+	if issuerSpec.Venafi != nil {
+		extKeyUsages = append(extKeyUsages, x509.ExtKeyUsageServerAuth)
+	}
+
+	// If using DNS01 ACME issuer with ECDSA then remove key encipherment
```

Comment:
I think we can remove the ACME parts here https://forums.developer.amazon.com/answers/61280/view.html